providers/amazon/src/airflow/providers/amazon/aws/operators/glue.py

```diff
@@ -231,7 +241,8 @@ def execute(self, context: Context):
                     run_id=self._job_run_id,
                     verbose=self.verbose,
                     aws_conn_id=self.aws_conn_id,
-                    job_poll_interval=self.job_poll_interval,
+                    waiter_delay=int(self.job_poll_interval),
+                    waiter_max_attempts=self.retry_limit,
```

Comment:
`self.retry_limit` is zero by default which means we don't ever attempt again and we'll fail immediately. This is causing our system tests to fail when run in deferrable mode:

![Screenshot from 2025-06-26 15-40-16](https://github.com/user-attachments/assets/e61947e5-6137-4744-a3ec-cf752369c3f2)